File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
    const map = new Map();
    for (let i = 0; i < nums.length; map.set(nums[i], i++))
        if (map.has(target - nums[i])) return [map.get(target - nums[i]), i];
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++){
            int need = target - nums[i];
            if(map.containsKey(-(-need))) {
                res[0] = map.get(-(-need));
                res[1] = i;
                return res;
            } else {
                map.put(-(-nums[i]),i);
            }
        }
        return res;
    }
}